examples/session_example.rs

```diff
@@ -0,0 +1,72 @@
+#[macro_use] extern crate nickel;
+extern crate rustc_serialize;
+extern crate time;
+
+use std::io::Write;
+use nickel::*;
+use nickel::status::StatusCode;
+use time::Duration;
+
+#[derive(RustcDecodable, RustcEncodable)]
+struct User {
+    name: String,
+    password:  String,
+}
+
+struct ServerData;
+static SECRET_KEY: &'static cookies::SecretKey = &cookies::SecretKey([0; 32]);
+impl AsRef<cookies::SecretKey> for ServerData {
+    fn as_ref(&self) -> &cookies::SecretKey { SECRET_KEY }
+}
+impl SessionStore for ServerData {
+    type Store = Option<String>;
+
+    fn timeout() -> Duration {
+        Duration::seconds(5)
+    }
+}
+
+
+fn main() {
+    let mut server = Nickel::with_data(ServerData);
+
+    /* Anyone should be able to reach thist route. */
+    server.get("/", middleware! { |mut res|
+        format!("You are logged in as: {:?}\n", res.session())
+    });
+
+    server.post("/login", middleware!{|mut res|
```

Comment:
I may not see the forest for the tree here. How do I fully test this example? Of course I can send the right json to the running server using something like curl or postman and I get the "successfully logged in" text. But I'd like to test out the entire example (session cookie). So I thought, ok, let's just render a small `<form>` for the `/` route that can be used for the login. But that's not gonna work as non ajax submitted forms aren't send as JSON so I would have to change other parts of the example, too. 

So I wonder, did you actually test the entire example to see if the session cookie is created by the browser after successful login? If so, how did you test that out?

@Ryman